src/olympia/devhub/templates/devhub/addons/listing/items.html

```diff
@@ -5,17 +5,20 @@
   <div class="item addon ignore-compatibility" data-addonid="{{ addon.id }}">
     <div class="info">
       {{ dev_heading(addon, amo) }}
-      {% if not addon.has_complete_metadata() and addon.status == amo.STATUS_NULL %}
+      {% if addon.status == amo.STATUS_NULL and not addon.has_complete_metadata() %}
         <p class="incomplete">
-          {% trans %}
-            This add-on will be deleted automatically after a few days if
-            the submission process is not completed.
-          {% endtrans %}
+          {{ _('This add-on is missing some required information before it can be submitted for publication.') }}
+        </p>
+      {% elif addon.status == amo.STATUS_NULL and not addon.has_unlisted_versions() and not addon.has_listed_versions() %}
+        <p class="incomplete">
+          {{ _("This add-on doesn't have any versions.")}}
         </p>
       {% else %}
-        <div class="item-info">
-          {{ dev_item_info(addon, amo) }}
-        </div>
+        {% if addon.status != amo.STATUS_NULL %}
+          <div class="item-info">
+            {{ dev_item_info(addon, amo) }}
+          </div>
+        {% endif %}
```

Comment:
mmm that part was mostly me being tired of seeing "0 ratings, 0 downloads, 0 active users" on unlisted add-ons, and then I figured it would be useless on incomplete add-ons as well... But I guess it's safer to hide it only for purely unlisted add-ons.